lib/oban/telemetry.ex

```diff
@@ -370,17 +374,19 @@ defmodule Oban.Telemetry do
 
   @doc false
   @spec handle_event([atom()], map(), map(), Logger.level()) :: :ok
-  def handle_event([:oban, :job, event], measure, meta, level) do
-    meta
-    |> Map.take([:args, :worker, :queue])
-    |> Map.merge(converted_measurements(measure))
-    |> log_message("job:#{event}", level)
-  end
-
-  def handle_event([:oban, :circuit, event], _measure, meta, level) do
-    meta
-    |> Map.take([:message, :name])
-    |> log_message("circuit:#{event}", level)
+  def handle_event(event_name, measure, meta, level) do
+    case Enum.slice(event_name, -2..-1) do
```

Comment:
I'm concerned about the performance of `Enum.slice` when processing a _lot_ of events. It's a bit manual, but we could specify clauses for prefixes with 1-3 values and put a limit in the `config` validation. Something like this

```elixir
[_, :job, event] -> # ...
[_, _, :job, event] -> # ...
[_, _, _, :job, event] -> # ...
```

That may be overkill, and I'm curious to see a benchmark that proves `Enum.slice` is good enough. WDYT?

Reply:
I was not sure about which was the best solution there. I'll try to come up with a benchmark

Reply:
Pattern matching is around 5-6x faster than relying on Enum.slice. 

I had two options: 
- defining a single clause and using a case
- defining 6 clauses

I went with the second one.